modules/rocksdb-common/src/main/java/org/apache/ignite/internal/rocksdb/flush/RocksDbFlusher.java

```diff
@@ -279,6 +281,18 @@ public void stop() {
      * @return Future that completes when the {@code onFlushCompleted} callback finishes.
      */
     CompletableFuture<Void> onFlushCompleted() {
-        return CompletableFuture.runAsync(onFlushCompleted, threadPool);
+        return inBusyLockSafeAsync(() -> runAsync(onFlushCompleted, threadPool));
+    }
+
+    private CompletableFuture<Void> inBusyLockSafeAsync(Supplier<CompletableFuture<Void>> supplier) {
```

Comment:
It doesn’t matter to me, I remember that we discussed it a long time ago in a slightly wider circle and came to the conclusion that such a convention is normal since we are returning the future.

Reply:
No, it is not a normal convention. We decided to give up because we couldn't come up with a better name. According to Ignite's conventions, and generally accepted conventions in Java, `async` methods do something asynchronously, which is not the case here

Reply:
What do you suggest?

Reply:
I suggest brainstorming a better name and rename all these methods in a separate JIRA, in order to avoid further confusion